### PR TITLE
Fixed #31426 -- Added proper field validation to QuerySet.order_by().

### PR DESCRIPTION
**Pull request by charettes**

### CODE REVIEW

File: tests/queries/tests.py [hunk 1]
```diff
@@ -3086,20 +3099,10 @@ def test_iter_exceptions(self):
...
-    def test_invalid_qs_list(self):
-        # Test for #19895 - second iteration over invalid queryset
-        # raises errors.
-        qs = Article.objects.order_by('invalid_column')
-        msg = "Cannot resolve keyword 'invalid_column' into field."
-        with self.assertRaisesMessage(FieldError, msg):
-            list(qs)
-        with self.assertRaisesMessage(FieldError, msg):
-            list(qs)
-
```
Comment (charettes) [inline tests/queries/tests.py:3098]: This test was now redundant; `FieldError` is raised at `order_by` call time and not at query compilation time anymore which is covered by the following test.

**Comment by charettes** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
@charettes Thanks :+1: I pushed minor edits.

**Comment by felixxm** [comment]
Docs failure is not related, see #12670.

**Comment by charettes** [comment]
Ahh was wondering what was wrong wrt docs!

**Pull request merged**